sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,221 @@
+        set at the same time as ``func`` and ``inverse_func``. If ``None`` and
+        ``func`` and ``inverse_func`` are ``None`` as well, the transformer
+        will be an identity transformer. The transformer will be cloned during
+        fitting.
+
+    func : function, optional
+        Function to apply to ``y`` before passing to ``fit``. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` is
+        ``None`` as well, the function used will be the identity function.
+
+    inverse_func : function, optional
+        Function to apply to the prediction of the regressor. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` as
+        well, the function used will be the identity function. The inverse
+        function is used to return to the same space of the original training
+        labels during prediction.
+
+    check_inverse : bool, (default=True)
+        Whether to check that ``transform`` followed by ``inverse_transform``
+        or ``func`` followed by ``inverse_func`` leads to the original data.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`. The random state is used for subsampling in
+        ``check_inverse``.
+
+    Attributes
+    ----------
+    regressor_ : object
+        Fitted regressor.
+
+    transformer_ : object
+        Used transformer in ``fit`` and ``predict``.
+
+    y_ndim_ : int
+        Number of targets.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.preprocessing import TransformTargetRegressor
+    >>> tt = TransformTargetRegressor(regressor=LinearRegression(),
+    ...                                 func=np.log, inverse_func=np.exp)
+    >>> X = np.arange(4).reshape(-1, 1)
+    >>> y = np.exp(2 * X).ravel()
+    >>> tt.fit(X, y)
+    ... #doctest: +NORMALIZE_WHITESPACE
+    TransformTargetRegressor(check_inverse=True,
+                             func=<ufunc 'log'>,
+                             inverse_func=<ufunc 'exp'>,
+                             random_state=None,
+                             regressor=LinearRegression(copy_X=True,
+                                                        fit_intercept=True,
+                                                        n_jobs=1,
+                                                        normalize=False),
+                             transformer=None)
+    >>> tt.score(X, y)
+    1.0
+    >>> tt.regressor_.coef_
+    array([ 2.])
+
+    Notes
+    -----
+    See examples/linear_model/plot_ols.py to highlight the benefit of
```

Comment:
format this like https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_validation.py#L767